test/programs/test_multi_server.py

```diff
@@ -8,33 +8,26 @@ def test_resolve(self) -> None:
         p.ctx.player_names = {
             (1, 1): "AAA",
             (1, 2): "aBc",
-            (1, 3): "abC",
+            (1, 3): "a bc",
         }
-        assert not p.resolve_player("abc"), "ambiguous name entry shouldn't resolve to player"
-        assert not p.resolve_player("Abc"), "ambiguous name entry shouldn't resolve to player"
-        assert p.resolve_player("aBc") == (1, 2, "aBc"), "matching case resolve"
-        assert p.resolve_player("abC") == (1, 3, "abC"), "matching case resolve"
-        assert not p.resolve_player("aB"), "partial name shouldn't resolve to player"
-        assert not p.resolve_player("abCD"), "incorrect name shouldn't resolve to player"
 
-        p.ctx.player_names = {
-            (1, 1): "aaa",
-            (1, 2): "abc",
-            (1, 3): "abC",
-        }
-        assert p.resolve_player("abc") == (1, 2, "abc"), "matching case resolve"
-        assert not p.resolve_player("Abc"), "ambiguous name entry shouldn't resolve to player"
-        assert not p.resolve_player("aBc"), "ambiguous name entry shouldn't resolve to player"
-        assert p.resolve_player("abC") == (1, 3, "abC"), "matching case resolve"
+        self.assertEqual(p.resolve_player("aBc"), (1, 2, "aBc"), "matching case resolve")
+        self.assertEqual(p.resolve_player("a_Bc"), (1, 3, "a bc"), "underscore should resolve to space")
+
+        self.assertFalse(p.resolve_player("aB"), "partial name shouldn't resolve to player")
+        self.assertFalse(p.resolve_player("abCD"), "incorrect name shouldn't resolve to player")
 
+        p = ServerCommandProcessor(Context("", 0, "", "", 0, 0, False))
```

Comment:
Note sure if you want to fix this, but this should probably be a separate test for readability.